### PR TITLE
Upgrade to SF4

### PR DESCRIPTION
**Pull request by epinxteren**
Migrate to sf4.

### CODE REVIEW

**Review by phavekes — APPROVED** [review]
Tested in our dev-enviroment, and it's still functioning as expected. I'll leave the code-nitpicking to @MKodde

File: homestead/after.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+sudo phpenmod -v 7.2 custom
+composer install
+xon
+echo 'cd /home/vagrant/code' >> ~/.profile
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | bash
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
+
+nvm install 8.9.1
```
Comment (MKodde) [inline homestead/after.sh:23]: This needs to be set to a Node 10 version to get the dev env to work as described in the development readme.

File: .travis.yml [hunk 2]
```diff
@@ -11,9 +11,7 @@ addons:
       - apache2
 
 php:
-  - 5.6
-  - 7.0
-  - 7.1
+  - 7.2
```
Comment (MKodde) [inline .travis.yml:14]: On line 23, in the befor_script section. NPM is installed at a pinned version. Maybe set this to the latest version 10 in order to get latest and greatest? Might save some additional update time in the future.

**Review by MKodde — COMMENTED** [review]
From my functional test, found two more findings. They both are touching on the Node/NPM version that is being installed.

**Comment by MKodde** [comment]
Came back from the dentist, did a functional review. I created a development environment as described in the `docs/development.md` file. That process was rather smooth, only the required NPM version was not installed. But I already mentioned that in a code review item above.

Some other findings:

1. On the `https://tiqr.example.com/demo/sp` page (should also be accessible via the StepUp VM box). I also had the certificate issue we discussed over the phone last week. 'app/config/key.pem' cannot be found. This should probably be fixed in the parameters.yml.dist file, and or in the Stepup-Deploy repo.
    1. We probably want these certs in the `config` folder and not in the deprecated `app` folder.
    2. The default values set in the dist file should be synced accordingly.
2. There are 44 deprecation warnings when visiting the `demo/sp` page. They should not pose any issues in this Symfony 4.4 version, but IMHO some effort should be taken to get rid of most of them. The ones that take more effort, or require additional development might have to wait for the next upgrade round. But in this effort you can ditch most of them with minimal effort.

**Comment by MKodde** [comment]
Feel free to merge this and fix some of the points on other PR's/stories. Maybe refer in this PR to those stories to keep things organized.

File: .travis.yml [hunk 3]
```diff
@@ -51,7 +51,7 @@ before_script:
 
   # Install dependencies
   - cp .env.ci .env
-  - cp config/packages/parameters.yaml.dist config/packages/parameters.yaml
+  - cp config/packages/parameters.yml.dist config/packages/parameters.yml
```
[the remark this answers is not in the document]
Comment (mroest) [reply .travis.yml:54]: To be honest. IMO we should squash this whole PR since the individuals commits are worthless.

**Comment by mroest** [comment]
> Came back from the dentist, did a functional review. I created a development environment as described in the `docs/development.md` file. That process was rather smooth, only the required NPM version was not installed. But I already mentioned that in a code review item above.
> 
> Some other findings:
> 
> 1. On the `https://tiqr.example.com/demo/sp` page (should also be accessible via the StepUp VM box). I also had the certificate issue we discussed over the phone last week. 'app/config/key.pem' cannot be found. This should probably be fixed in the parameters.yml.dist file, and or in the Stepup-Deploy repo.
>    
>    1. We probably want these certs in the `config` folder and not in the deprecated `app` folder.
>    2. The default values set in the dist file should be synced accordingly.
> 2. There are 44 deprecation warnings when visiting the `demo/sp` page. They should not pose any issues in this Symfony 4.4 version, but IMHO some effort should be taken to get rid of most of them. The ones that take more effort, or require additional development might have to wait for the next upgrade round. But in this effort you can ditch most of them with minimal effort.

yes, please open additional stories for this since its unrelated to this PR.

**Comment by MKodde** [comment]
Creating issues for 1.1 and 1.2 is fine with me, but 2 might be part of the SF4 upgrade. Having an installation without excessive deprecation warnings is part of the upgrade. But I'm fine with addressing them on a successive PR to prevent adding to this big change set.

Are you okay with that proposal?

**Comment by mroest** [comment]
> Creating issues for 1.1 and 1.2 is fine with me, but 2 might be part of the SF4 upgrade. Having an installation without excessive deprecation warnings is part of the upgrade. But I'm fine with addressing them on a successive PR to prevent adding to this big change set.
> 
> Are you okay with that proposal?

Agree that it should clutter logs with warnings but, the demo/sp is only used for testing and I've checked the deprecation warnings but they are actually from vendor packages?

**Comment by mroest** [comment]
As discussed with @MKodde we will create additional stories to resolve deprecation warnings.

**Pull request merged**